Source/Dafny/Compilers/Compiler-python.cs

```diff
@@ -686,7 +685,21 @@ protected class ClassWriter : IClassWriter {
 
     protected override void EmitNewArray(Type elmtType, IToken tok, List<Expression> dimensions, bool mustInitialize,
       ConcreteSyntaxTree wr, ConcreteSyntaxTree wStmts) {
-      throw new NotImplementedException();
+      var initValue = mustInitialize ? DefaultValue(elmtType, wr, tok, true) : "None";
+      if (dimensions.Count == 1) {
+        // handle the common case of 1-dimensional arrays separately
+        var arrayType = TypeInitializationValue(elmtType, wr, tok, false, false);
+        wr.Write($"[{initValue} for _ in range");
+        TrParenExpr(dimensions[0], wr, false, wStmts);
+        wr.Write("]");
+      } else {
+        wr.Write("_dafny.newArray({0}", initValue ?? "None");
+        foreach (var dim in dimensions) {
+          wr.Write(",int");
```

Comment:
I don't think that we need those `int`s, but if you insist on them, please get the spacing right.